### DISCUSSION TITLE
How is it best to provide a Kotlin extension?

### DISCUSSION THREAD

**Discussion by vladimirfx** [ebean-orm/ebean#2397] [General]
In our projects, we accumulate some universal and useful Kotlin extension functions that simplify/canonize data access code. 
For example, reified nonnull find by id:

```Kotlin

inline fun <reified T> Database.getById(id: Any): T {
    return checkNotNull(find(T::class.java, id)) {
        "Entity ${T::class.java} with id $id does not exists in DB."
    }
}

```

How it can be contributed to the Ebean project?

**Comment by rbygrave** [comment]
Ah missed this.  We can create a "kotlin extensions module" that we could put these into?

**Reply by vladimirfx** [reply]
Maybe name it `ebean-kotlin` ? 
In addition to API extensions, we cant put into it special support for value classes and coroutine-based transaction manager/context.

Please create that module and I provide first extensions for it.

**Comment by rbygrave** [comment]
Ok, we have a ebean-kotlin module now ... https://github.com/ebean-orm/ebean/issues/2451
